storage/database/db_storages/bill_storage.go

```diff
@@ -57,11 +58,57 @@ func (b *BillStorage) GetByID(id uuid.UUID) (BillModel, error) {
 func (b *BillStorage) CreateItem(item ItemModel) (ItemModel, error) {
 	itemEntity := ToItemEntity(item)
 
+	// TODO: if userId belongs to deleted user do not create
 	// store item
 	res := b.DB.Create(&itemEntity)
+
+	// TODO: check if other errors can occur
+	if res.Error != nil {
+		return ItemModel{}, storage.NoSuchUserError
+	}
 	if res.RowsAffected == 0 {
-		return ItemModel{}, storage.BillAlreadyExistsError
+		return ItemModel{}, storage.ItemAlreadyExistsError
 	}
 
-	return ToItemModel(itemEntity), res.Error
+	return ToItemModel(itemEntity), nil
+}
+
+func (b *BillStorage) GetItemByID(id uuid.UUID) (ItemModel, error) {
+	var item Item
+	tx := b.DB.Preload("Contributors").Limit(1).Find(&item, "id = ?", id)
+	if tx.RowsAffected == 0 {
+		return ItemModel{}, storage.NoSuchItemError
+	}
+	return ToItemModel(item), nil
+}
+
+func (b *BillStorage) UpdateItem(item ItemModel) (ItemModel, error) {
+	itemEntity := ToItemEntity(item)
+
+	// run as a transaction to ensure consistency. item should be completely updated or not at all
+	err := b.DB.Transaction(func(tx *gorm.DB) error {
+		// update contributors associations
+		res := tx.
+			Model(&itemEntity).
+			Association("Contributors").
+			Replace(itemEntity.Contributors)
+
+		// TODO: add finer error handling
+		if res != nil {
+			return storage.NoSuchUserError
+		}
+
+		// update base item fields
+		ret := tx.
+			Model(&itemEntity).
+			Updates(&itemEntity)
+
+		if ret.Error != nil {
+			return storage.NoSuchItemError
+		}
```

Comment:
Here we need to check how many rows are affected: If ret.RowsAffected == 0 -> Item with the given ID does not exist. So we need to call an NoSuchItemErr.

Also i would first update the items base fields and afterwards the associations. This will make no difference but i think it will improve the code readability.

Reply:
done